threescale_api/resources.py

```diff
@@ -534,7 +534,10 @@ def url(self) -> str:
         return self.parent.url + '/oidc_configuration'
 
     def update(self, params: dict = None, **kwargs) -> 'DefaultResource':
-        return self.rest.patch(url=self.url, json=params, **kwargs)
+        return self.rest.patch(url=self.url, json=params, **kwargs).json()
```

Comment:
according to type annotation this should return DefaultResource, not json. Furthermore if json is returned, information abot status (http code) is lost forever, impossible to say whether passed or failed

Reply:
@mganisin Ok, let's change also read function and union the behavior. There is no response in form of `DefaultResource`, but `requests.Response`. Do you agree?

Reply:
I've change it.

Reply:
this still claims it should return 'DefaultResource' but returns json. Type annotation and return value should be in match, it should be both 'DefaultResource' or json (is it Dict?]

same applies to `def read` below

Reply:
@mganisin this should be resolved too